python/tvm/script/parse/tir/tir.py

```diff
@@ -22,31 +22,61 @@
 from tvm.tir import Buffer, IterVar, PrimExpr, Var
 from tvm.ir import PrimType
 
-from ...builder import Frame, def_
+from ...builder import Frame, name
 from ...builder import tir as T
 from .. import dispatch, doc
 from ..parser import Parser
 
 
-def bind_value(self: Parser, name: str, value: Any) -> Any:
-    if isinstance(value, Frame):
+def bind_with_value(self: Parser, var_name: str, value: Any) -> Any:
+    if isinstance(value, (list, tuple)):
+        for i in range(len(value)):
+            bind_with_value(self, f"{var_name}_{i}", value[i])
+        return value
+    elif isinstance(value, (Buffer, Var)):
+        name(var_name, value)
+        return value
+    else:
+        self.report_error(f"Do not know how to bind type: {type(value)} in with statement")
```

Comment:
Note: there is still a missing argument in `self.report_error`

Reply:
yes, but we do not have `node` in `bind_value`